src/edu/stanford/nlp/sempre/ParserState.java

```diff
@@ -227,6 +227,58 @@ protected void pruneCell(String cellDescription, List<Derivation> derivations) {
     return derivs;
   }
 
+  public List<Derivation> gatherTokenAndPhraseDerivations(int spanStart, int spanEnd) {
```

Comment:
Seems like this only depends on Example - make this static function and pass in Example for clarity.  Document what this function does.